core/src/avm2/globals/json.rs

```diff
@@ -277,21 +289,35 @@ pub fn stringify<'gc>(
     _this: Option<Object<'gc>>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    let val = args.get(0).unwrap_or(&Value::Undefined);
-    let replacer = args.get(1).unwrap_or(&Value::Undefined).as_object();
-    let spaces = args.get(2).unwrap_or(&Value::Undefined);
+    let val = args.get_value(0);
+    let replacer = args.get_value(1).as_object();
+    let spaces = args.get_value(2);
+
+    // If the replacer is None, that means it was a primitive (as_object returns None for primitives), and therefore not a valid replacer
+    if replacer.is_none() && !matches!(args.get(1).unwrap(), Value::Null) {
```

Comment:
This then doesn't need the second half of the condition (null test)

Reply:
This causes tests to fail because then a null replacer becomes invalid, and the default replacer is null.

Reply:
I don't understand, `try_get_object` should return none for null